go/integration_test/processor/calls_test.go

```diff
@@ -176,37 +171,274 @@ func TestProcessor_DifferentCallTypesAccessStorage(t *testing.T) {
+						t.Errorf("%s did forward value, got unexpected output: %v", callName, result.Output)
+					}
+				}
+			})
+		}
+	}
+}
+
+func TestProcessor_DifferentCallTypesSetTheCorrectSender(t *testing.T) {
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue // todo implement different call types
+		}
+		for callName, call := range callTypesAndProperties() {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+				sender0 := tosca.Address{1}
+				receiver0 := tosca.Address{2}
+				receiver1 := tosca.Address{3}
+
+				// set call arguments
+				code0 := pushCallArguments(call, sufficientGas, tosca.Value{}, receiver1)
+				// perform call and forward the result
+				code0 = append(code0, []byte{
+					byte(call.callType),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				// inner call, get caller and return it
+				code1 := []byte{
+					byte(vm.CALLER),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}
+
+				state := WorldState{
+					sender0:   Account{},
+					receiver0: Account{Code: code0},
+					receiver1: Account{Code: code1},
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender0,
+					Recipient: &receiver0,
+					GasLimit:  sufficientGas,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Errorf("execution was not successful or failed with error %v", err)
+				}
+				if call.sameSender {
+					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 12), sender0[:]...)) {
+						t.Errorf("%s did not set the correct sender, got unexpected output: %v", callName, result.Output)
+					}
+				} else {
+					if !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 12), receiver0[:]...)) {
+						t.Errorf("%s did not set the correct sender, got unexpected output: %v", callName, result.Output)
 					}
 				}
 			})
 		}
 	}
 }
 
+func TestProcessor_RecursiveCallsAfterAStaticCallAreStatic(t *testing.T) {
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue // todo implement different call types
+		}
+		calls := callTypesAndProperties()
+		for callName, call := range calls {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+
+				// Test structure:
+				// sender0 --CALL--> receiver0 --STATICCALL--> receiver1
+				// --test.CALL--> receiver2 -> SSTORE
+				// after the first static calls all following calls have to be static,
+				// therefore the SSTORE should revert. The success boolean of the test.CALL
```

Comment:
A thing about writing: should is conditional, this means that not doing it is acceptable.
- Use shall when you want to be specific about intention. 
- Use may when it is intentionally conditional. 

 (requirements texts avoid using must, will, could, and should because of different semantics)